src/main/java/com/minecolonies/client/gui/WindowHutGuardTower.java

```diff
@@ -0,0 +1,467 @@
+     * @param button the button clicked to switch the task.
+     */
+    private void switchTask(final Button button)
+    {
+        if(button.getID().contains("patrol"))
+        {
+            building.task = BuildingGuardTower.Task.PATROL;
+
+            buttonTaskPatrol.setEnabled(false);
+            buttonTaskFollow.setEnabled(true);
+            buttonTaskGuard.setEnabled(true);
+
+            buttonSetTarget.show();
+        }
+        else if(button.getID().contains("follow"))
+        {
+            building.task = BuildingGuardTower.Task.FOLLOW;
+
+            buttonTaskFollow.setEnabled(false);
+            buttonTaskPatrol.setEnabled(true);
+            buttonTaskGuard.setEnabled(true);
+
+            buttonSetTarget.hide();
+        }
+        else
+        {
+            building.task = BuildingGuardTower.Task.GUARD;
+
+            buttonTaskGuard.setEnabled(false);
+            buttonTaskPatrol.setEnabled(true);
+            buttonTaskFollow.setEnabled(true);
+
+            buttonSetTarget.show();
+        }
+        pullInfoFromHut();
+        sendChangesToServer();
+    }
+
+    /**
+     * Sets the target for patrolling or guarding of the guard.
+     * @param button clicked button.
+     */
+    private void setTarget(final Button button)
+    {
+        final EntityPlayerSP player = this.mc.thePlayer;
+        final int emptySlot = player.inventory.getFirstEmptyStack();
+        pullInfoFromHut();
+
+        if(emptySlot == -1)
+        {
+            LanguageHandler.sendPlayerLocalizedMessage(player, "com.minecolonies.gui.workerHuts.noSpace");
+        }
+
+        if(patrolManually && task.equals(BuildingGuardTower.Task.PATROL))
+        {
+            givePlayerScepter(BuildingGuardTower.Task.PATROL);
+            LanguageHandler.sendPlayerLocalizedMessage(player, "com.minecolonies.job.guard.tool.taskPatrol");
+        }
+        else if(task.equals(BuildingGuardTower.Task.GUARD))
+        {
+            givePlayerScepter(BuildingGuardTower.Task.GUARD);
+            LanguageHandler.sendPlayerLocalizedMessage(player, "com.minecolonies.job.guard.tool.taskGuard");
+        }
+        window.close();
+    }
+
+
+    /**
+     * Send message to player to add scepter to his inventory.
+     * @param localTask the task to execute with the scepter.
+     */
+    private void givePlayerScepter(BuildingGuardTower.Task localTask)
+    {
+        MineColonies.getNetwork().sendToServer(new GuardScepterMessage(localTask.ordinal(), building.getID()));
+    }
+
+    /**
+     * Sends changes to the server.
+     */
+    private void sendChangesToServer()
+    {
+        MineColonies.getNetwork().sendToServer(new GuardTaskMessage(building, job.ordinal(), assignManually, patrolManually, retrieveOnLowHealth, task.ordinal()));
+    }
+
+    /**
+     * Switch the retrieval mode.
+     * @param button clicked button
+     */
+    private void switchRetrievalMode(final Button button)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "button". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1172)